x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/filestructurefinder/FileStructureFinderManager.java

```diff
@@ -474,14 +474,17 @@ FileStructureFinder makeBestStructureFinder(List<String> explanation, String sam
         Character quote = overrides.getQuote();
         Boolean shouldTrimFields = overrides.getShouldTrimFields();
         List<FileStructureFinderFactory> factories;
+        double allowedFractionOfBadLines = 0.0;
         if (delimiter != null) {
+            allowedFractionOfBadLines = DelimitedFileStructureFinderFactory.DEFAULT_BAD_ROWS_PERCENTAGE;
 
             // If a precise delimiter is specified, we only need one structure finder
             // factory, and we'll tolerate as little as one column in the input
             factories = Collections.singletonList(new DelimitedFileStructureFinderFactory(delimiter, (quote == null) ? '"' : quote, 1,
                 (shouldTrimFields == null) ? (delimiter == '|') : shouldTrimFields));
 
         } else if (quote != null || shouldTrimFields != null) {
+            allowedFractionOfBadLines = DelimitedFileStructureFinderFactory.DEFAULT_BAD_ROWS_PERCENTAGE;
```

Comment:
I think we should have wiggle room here as well. Because the user supplying these fields implies they expect it to be delimited.

Reply:
If we do it here then we should also do it for the `else` branch below if `format` has been overridden to `delimited`.  (The easiest way to do this might be to change the condition above to `else if (quote != null || shouldTrimFields != null || overrides.getFormat() == Format.DELIMITED)`.)

This does cause more problems though, for example, if a file has 12 semi-colons on all lines and 3 commas on 90% of lines we'll say it's CSV because we check that first.  It's probably OK, but we may want to tolerate fewer mismatches in this branch than the branch where the exact delimiter was specified.  (See my other comment about naming the constant.)